sci-libs/cantera/cantera-2.6.0-r2.ebuild

```diff
@@ -0,0 +1,151 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{8..11} )
+
+FORTRAN_NEEDED=fortran
+FORTRAN_STANDARD="77 90"
+
+inherit fortran-2 python-single-r1 scons-utils toolchain-funcs
+
+DESCRIPTION="Object-oriented tool suite for chemical kinetics, thermodynamics, and transport"
+HOMEPAGE="https://www.cantera.org"
+SRC_URI="https://github.com/Cantera/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+cti fortran lapack +python test"
+RESTRICT="!test? ( test )"
+
+REQUIRED_USE="
+	python? ( cti )
+	${PYTHON_REQUIRED_USE}
+"
+
+RDEPEND="
+	${PYTHON_DEPS}
+	lapack? ( virtual/lapack )
+	cti? (
+		$(python_gen_cond_dep '
+			dev-python/ruamel-yaml[${PYTHON_USEDEP}]
+		')
+	)
+	python? (
+		$(python_gen_cond_dep '
+			dev-python/numpy[${PYTHON_USEDEP}]
+		')
+	)
+	dev-cpp/yaml-cpp
+	!lapack? ( sci-libs/sundials:0= )
+	lapack? ( >=sci-libs/sundials-6.5.0:0=[lapack?] )
```

Comment:
Is this minimum version dep only grounded in test failure? If so, I don't think it is necessary, unless upstream makes a recommendation.

- ~arch is fine by updating to latest sundials version
- arch will be fixed by 6.5.0 stabilisation and subsequent cleanup old

Reply:
The problem is that upstream never tested cantera 2.6.0 with sundials 6.2, only with 5.2, 5.8, 6.0 and with gcc-9 as I can see. Maybe it related with gcc version too (and it's additional gentoo patches). In my 2 systems the <=sundials-6.2[lapack,int64] fails tests itself for gcc-11 but not gcc-12 while sundials-6.5 is ok for both gcc versions.

I reported the issue to upstream but I currently try to reproduce it on Ubuntu with sundials as submodule because Ubuntu package for sundials built without lapack support. It takes more time than I expected. I see the cantera[lapack] tests failures for current upstream too in my system.

Actually the lapack support is alternative  for eigen for several tasks. It's not the full replacement. The eigen is always required. Initially gentoo cantera package was without it.